core-web/libs/portlets/edit-ema/portlet/src/lib/edit-ema-editor/edit-ema-editor.component.ts

```diff
@@ -284,6 +266,32 @@ export class EditEmaEditorComponent implements OnInit, OnDestroy {
         // this.store.updateEditorState(EDITOR_STATE.LOADING);
     }
 
+    /**
+     * Handles the reload of content in the editor.
+     * If the editor state is LOADED and the content is not VTL, it reloads the iframe.
+     * If the content is VTL, it loads the VTL iframe content.
+     * @memberof EditEmaEditorComponent
+     */
+    handleReloadContent() {
+        this.store.stateLoad$
+            .pipe(
+                takeUntil(this.destroy$),
+                filter((editorState) => editorState === EDITOR_STATE.LOADED),
+                switchMap(() => this.store.code$)
+            )
+            .subscribe((code) => {
+                if (!this.isVTLPage()) {
+                    // Only reload if is Headless.
+                    // If is VTL, the content is updated by store.code$
+                    this.reloadIframe();
+
+                    return;
+                }
+
```

Comment:
We can reset the dialog here, remember to add the logic to prevent the reset if the dialog is hidden.

Reply:
It seems to me that we agreed to handle that directly in the callback, because we don't always want to close the dialog when the editor state becomes "Loaded"

Reply:
Yes you are right. I forgot that